src/lib/bootloader/grub2base.rb

```diff
@@ -125,6 +135,39 @@ def disable_serial_console
 
   private
 
+    def merge_pmbr_action(other)
+      @pmbr_action = other.pmbr_action if other.pmbr_action
+    end
+
+    def merge_sections(other)
+      sections.default = other.sections.default if other.sections.default
+    end
+
+    def merge_password(other)
+      @password = other.password
+    end
+
+    def merge_grub_default(other)
+      default = grub_default
+      other = other.grub_default
+
+      unless other.kernel_params.serialize.empty?
+        new_kernel_params = default.kernel_params.serialize + " " + other.kernel_params.serialize
+        default.kernel_params.replace(new_kernel_params)
+      end
+
+      # string attributes
+      [:serial_console, :terminal, :timeout, :hidden_timeout, :distributor,
+       :gfx_mode, :theme].each do |attr|
+        default.send((attr.to_s + "="), other.send(attr)) if other.send(attr)
+      end
+
+      # boolean attributes
+      [:os_prober].each do |attr|
+        default.send(attr).value = other.send(attr).enabled? if other.send(attr).defined?
```

Comment:
I wondered whether `defined?` was a mistake until I realized that `default` and `other` were [CFA::BooleanValue](http://www.rubydoc.info/github/config-files-api/config_files_api/master/CFA/BooleanValue). Mention that in a comment?